frontends/web/src/utils/request.ts

```diff
@@ -112,3 +112,14 @@ export function apiPost(
     body: JSON.stringify(body)
   }).then(response => response.json()).then(handleError(endpoint));
 }
+
+export function generateQueryString<T extends Record<string, unknown>>(params: T): string {
+  const filteredAndSortedParams = Object.keys(params)
+    .filter(key => params[key] !== null && params[key] !== undefined) // filters out null & undefined values
+    .reduce((acc, key) => {
+      acc[key] = params[key];
+      return acc;
+    }, {} as Record<string, any>); // constructs a new object with the keys
+
+  return new URLSearchParams(filteredAndSortedParams).toString();
```

Comment:
I'd love if we could just  use `new URLSearchParams(obj).toString();` instead of Object.keys, filter, reduce.

but this includes null/undefined, as they are part of the object. Is it breez that expects the absence of the params or is it the BBApp lightning implementation in BBApp?

Reply:
yes, id love to just use URLSearchParams directly too if that fits our requirements. The "complexity" comes from the filtering and re-building the object.

> Is it breez that expects the absence of the params or is it the BBApp lightning implementation in BBApp?

Not sure. cc @dangeross

Reply:
Filtering has to be done on the frontend or backend before being sent to the SDK. Removing it on the frontend means the backend needs to handle it. Maybe you could change [these go functions](https://github.com/dangeross/bitbox-wallet-app/blob/staging-ln-framework/backend/lightning/util.go) (`getOptional...`) to handle empty string values

Reply:
> Maybe you could change [these go functions](https://github.com/dangeross/bitbox-wallet-app/blob/staging-ln-framework/backend/lightning/util.go) (getOptional...) to handle empty string values

cc @Beerosagos wdyt?

Reply:
Yes, I agree. Better to handle it in the backend, and the go functions indicated by @dangeross sound good :+1:

Reply:
Sounds good. We can directly use new `URLSearchParams(obj).toString`. 

> Maybe you could change [these go functions](https://github.com/dangeross/bitbox-wallet-app/blob/staging-ln-framework/backend/lightning/util.go) (getOptional...) to handle empty string values

Just one question, should we then pass the undef/null values as empty strings? 🙂

Reply:
I'd say you can pass them as it is easier for you. We can filter appropriately in the backend

Reply:
Yeah, you should be able to pass empty strings and the backend should prepare the request to the SDK. If there is a mandatory key missing/empty, it should throw an error.